fs/ftpfs.py

```diff
@@ -317,17 +317,39 @@ def __str__(self):
         )
         return _fmt.format(host=self.host, port=self.port)
 
-    def _open_ftp(self, encoding="utf-8"):
+    def _open_ftp(self):
         _ftp = FTP()
         _ftp.set_debuglevel(0)
         with ftp_errors(self):
-            _ftp.encoding = encoding
             _ftp.connect(self.host, self.port, self.timeout)
             _ftp.login(self.user, self.passwd, self.acct)
+            self._features = {}
+            try:
+                feat_response = _decode(_ftp.sendcmd("FEAT"), 'latin-1')
+            except error_perm:
+                self.encoding = 'latin-1'
+            else:
+                if feat_response.split('-')[0] == '211':
+                    for line in feat_response.splitlines():
+                        if line.startswith(' '):
+                            k, _, v = line[1:].partition(' ')
+                            self._features[k] = v
+                self.encoding = (
+                    'utf-8'
+                    if 'UTF8' in self._features
+                    else 'latin-1'
+                )
+                if not PY2:
+                    _ftp.file = _ftp.sock.makefile(
```

Comment:
Not sure how much of a hack this is. The `file` attribute of FTP objects is the only thing that references the encoding, by re-creating it, we can modify the encoding based on features.

Reply:
It should be fine, I don't think the `ftplib` implementation will change since it is part of the stdlib.

Isn't your linter complaining about the seemingly-useless `self.ftp` lines in other methods though ?

Reply:
> It should be fine, I don't think the ftplib implementation will change since it is part of the stdlib.

That's my thinking.

> Isn't your linter complaining about the seemingly-useless self.ftp lines in other methods though ?

It is, and a few more complaints besides. I'll see if I can clear some of those up.